src/background.c

```diff
@@ -222,12 +288,30 @@ handle_request_background_in_thread_func (GTask *task,
   else
     allowed = permission == YES ? TRUE : FALSE;
 
+  g_debug ("Setting autostart for %s to %s", app_id,
+           allowed && autostart_requested ? "enabled" : "disabled");
+
+  commandline = rewrite_commandline (app_id, autostart_exec);
```

Comment:
We should not rewrite if xdp_app_info_is_host(). In fact, things are probably nicer if we added a xdg_app_info_rewrite_commandline() function. Then we can plug in e.g. snap support later.